### DISCUSSION TITLE
When i use prisma to fetch from my mysql db, my next js App and children component renders 2 times

### DISCUSSION THREAD

**Discussion by iamshimer** [prisma/prisma#12664] [Q&A]
### Bug description

I use prisma to fetch data from mySql workbench 8.0. I followed the correct order of implementation to integrate into my next js project with the help of your docs and some YT videos. But in the end i fetch data successfully, But the problem is that when i fetch data with the use effect and it makes my App component and it's children re-render even i use React.memo()

### How to reproduce

1. Run mysql server and create a database called mydb
2. clone this repo [https://github.com/iamshimer/nextJs-mysql-prisma.git](url)
3. Replace your mysql username, password and port in env file
4. Run npx prisma migrate dev
5. run **npm run dev**
6. see the console log of your browser

### Expected behavior

It will start and run successfully, But You can see the will be 3 pair of logs in console. Each of them contains a duplicate one. See my console
![Screenshot_20](https://user-images.githubusercontent.com/45072945/161361443-a2202447-a69e-428c-9346-54558b90fadb.png)

### Prisma information

``` 
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model User {
  id        Int    @id @default(autoincrement())
  email     String @db.VarChar(255)
  password  String @db.VarChar(255)
  firstName String @map("first_name") @db.VarChar(255)
  lastName  String @map("last_name") @db.VarChar(255)
}
```

### Environment & setup

- OS:  Windows
- Database: MySQL
- Node.js version: 16.13.0

### Prisma Version

```
3.11.1
```

**Comment by austincrim** [comment]
Hey @iamshimer 👋 ,

This could be due to [`React.StrictMode`](https://reactjs.org/docs/strict-mode.html#detecting-unexpected-side-effects) which causes double renders in development mode in order to detect unwanted side effects.

From the React docs:
> In React 17, React automatically modifies the console methods like console.log() to silence the logs in the second call to lifecycle functions. However, it may cause undesired behavior in certain cases where [a workaround can be used](https://github.com/facebook/react/issues/20090#issuecomment-715927125). Starting from React 18, React does not suppress any logs. However, if you have React DevTools installed, the logs from the second call will appear slightly dimmed. React DevTools also offers a setting (off by default) to suppress them completely.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
